test/AutoRest.Shared.Tests/ChangeTrackingListTest.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    public class ChangeTrackingListTest
+    {
+        [Test]
+        public void UndefinedByDefault()
+        {
+            var list = new ChangeTrackingList<string>();
+            Assert.True(list.IsUndefined);
+        }
+
+        [Test]
+        public void ReadOperationsDontChange()
+        {
+            var list = new ChangeTrackingList<string>();
+            _ = list.Count;
+            _ = list.IsReadOnly;
+            _ = list.Contains("a");
+            _ = list.IndexOf("a");
+            _ = list.Remove("a");
+
+            foreach (var kvp in list)
+            {
+            }
+
+            list.CopyTo(new string[5], 0);
+
+            Assert.Throws<IndexOutOfRangeException>(() => _ = list[0]);
+
+            Assert.True(list.IsUndefined);
+        }
+
+        [Test]
+        public void CanAddElement()
```

Comment:
Tests are missing for some operations like GetEnumerator(), Insert(), contains()...etc

Reply:
GetEnumerator is covered by foreach. You are right about the others. Adding.